rotkehlchen/db/calendar.py

```diff
@@ -0,0 +1,223 @@
+from typing import TYPE_CHECKING, Any, NamedTuple
+
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.db.filtering import (
+    DBFilter,
+    DBFilterQuery,
+    DBMultiIntegerFilter,
+    DBMultiStringFilter,
+    DBSubStringFilter,
+    DBTimestampFilter,
+    FilterWithTimestamp,
+)
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.types import BlockchainAddress, SupportedBlockchain, Timestamp
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+class CalendarRow(NamedTuple):
+    """Represents a calendar entry in the database"""
+    identifier: str
+    name: str
+    timestamp: Timestamp
+    description: str | None
+    counterparty: str | None
+    address: BlockchainAddress | None
+    blockchain: SupportedBlockchain | None
+
+    def serialize(self) -> dict[str, Any]:
+        return {
+            'identifier': self.identifier,
+            'name': self.name,
+            'description': self.description,
+            'counterparty': self.counterparty,
+            'timestamp': self.timestamp,
+            'address': self.address,
+            'blockchain': self.blockchain.serialize() if self.blockchain is not None else None,
+        }
```

Comment:
I asked you to add a serialize_for_db() and deserialize_from_db method for this and use it below in the various parts in this file and elsewhere (if needed)